casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -215,6 +217,13 @@ object ProtoUtil {
       ProtoUtil.unsafeGetBlock[F](parentHash)
     }
 
+  // TODO: Ensure deploy.get and raw.get are safe
```

Comment:
What about this **TODO** is this for the future Kent to solve?

Reply:
I understand Kent's pain that the optional fields in the proto classes are horrible to deal with but I totally agree with Mateusz and we should properly handle the missing field now, not in the future.

We should either getOrElse like in the code below, if that's a good default, or pass the responsibility of choosing such a default to the caller by changing the return type of the function to Optional[Boolean]

```scala
  def containsDeploy(b: BlockMessage, user: ByteString, timestamp: Long): Boolean = {
    deploys(b)
      .toStream
      .traverse(getDeployData)
      .map(_.exists(deployData => deployData.user == user && deployData.timestamp == timestamp))
      .getOrElse(false)
    
  def getDeployData(d : ProcessedDeploy) =
    for {
      deploy <- d.deploy
      deployData <- deploy.raw
    } yield  deployData
  }
```